File: src/main/js/funk/tuple/Tuple1.js
funk.tuple = funk.tuple || {};
funk.tuple.Tuple1 = (function(){
    "use strict";
    var TupleImpl = function(_1){
        this.__1 = _1;
    };
    TupleImpl.prototype = new funk.tuple.Tuple();
    TupleImpl.prototype.constructor = TupleImpl;
    TupleImpl.prototype._1 = function(){
        return this.__1;
    };
    TupleImpl.prototype.productArity = function(){
        return 1;
    };
    TupleImpl.prototype.productElement = function(index){
        switch(index) {
            case 0: return this._1();
            default: throw new funk.error.RangeError();
        }
    };
    TupleImpl.prototype.productPrefix = function(){
        return "Tuple1";
    };
    TupleImpl.prototype.name = "Tuple1";
    return TupleImpl;
})();

funk.tuple.tuple1 = function(_1){
    return new funk.tuple.Tuple1(_1);
};

// Alias
var tuple1 = funk.tuple.tuple1;

File: src/main/dart/funk/tuple/Tuple1.dart
interface Tuple1<A> extends ITuple default TupleImpl1 {
  A get _1();
}

class TupleImpl1<A> extends Product implements Tuple1<A> {
  
  A __1;
  
  TupleImpl1(A _1) {
    __1 = _1;
  }
  
  A get _1() => __1;
  
  A productElement(int index) {
    if(index >= productArity) {
      throw new RangeError([]);
    }
    return _1;
  }

  int get productArity() => 1;

  String get productPrefix() => "Tuple";
}

tuple1(_1) => new TupleImpl1(_1); 